File: TypeScript/ProjectEuler7.ts


class ListGenerator {
    private list: number[] = [];

    constructor(to: number) {
        var number1 = 2;
        var counter = 0;
        while (counter < to) {
            if (this.isPrime(number1)) {
                this.list.push(number1);
                counter++;
            }
            number1++;
        }
    }

    private isPrime = (numberr: number) => this.generateList(2, Math.sqrt(numberr)).filter(x=> numberr % x == 0).length == 0

    private generateList = function (start, stop) {
        var result: number[] = [];
        for (var i = start; i <= stop; i++) {
            result.push(i);
        }
        return result;
    }

    public getList = () => this.list;
}

var list = new ListGenerator(10000);

console.log(list.getList()[list.getList().length - 1])

while (true) { }

File: Python/ProjectEuler7.py

import math

def isPrime(number):
   return len([x for x in range(2,int(math.sqrt(number))+1) if number % x == 0 ]) == 0

def generator(first):
    def wrap(func,filterFunc):
        def seq(to):
            x = first
            counter = 0 
            while counter < to:
                if filterFunc(x):
                    counter = counter + 1
                    yield x
                x = func(x)
        return seq
    return wrap

result = list(generator(2)(lambda x: x + 1,lambda x: isPrime(x))(10000))

print(result[::-1][0])
